pkg/providers/subnet/subnet.go

```diff
@@ -96,20 +104,24 @@ func (p *DefaultProvider) List(ctx context.Context, nodeClass *v1beta1.EC2NodeCl
 		if err != nil {
 			return nil, fmt.Errorf("describing subnets %s, %w", pretty.Concise(filters), err)
 		}
+		availableZones := map[string]string{}
 		for i := range output.Subnets {
 			subnets[lo.FromPtr(output.Subnets[i].SubnetId)] = output.Subnets[i]
 			p.availableIPAddressCache.SetDefault(lo.FromPtr(output.Subnets[i].SubnetId), lo.FromPtr(output.Subnets[i].AvailableIpAddressCount))
 			p.associatePublicIPAddressCache.SetDefault(lo.FromPtr(output.Subnets[i].SubnetId), lo.FromPtr(output.Subnets[i].MapPublicIpOnLaunch))
 			// subnets can be leaked here, if a subnets is never called received from ec2
 			// we are accepting it for now, as this will be an insignificant amount of memory
 			delete(p.inflightIPs, lo.FromPtr(output.Subnets[i].SubnetId)) // remove any previously tracked IP addresses since we just refreshed from EC2
+			// add zone name to ID comparison as subnets are discovered
+			availableZones[aws.StringValue(output.Subnets[i].AvailabilityZone)] = aws.StringValue(output.Subnets[i].AvailabilityZoneId)
 		}
+		p.zoneInfo = availableZones
 	}
 	p.cache.SetDefault(fmt.Sprint(hash), lo.Values(subnets))
 	if p.cm.HasChanged(fmt.Sprintf("subnets/%s", nodeClass.Name), subnets) {
 		log.FromContext(ctx).
 			WithValues("subnets", lo.Map(lo.Values(subnets), func(s *ec2.Subnet, _ int) string {
-				return fmt.Sprintf("%s (%s)", aws.StringValue(s.SubnetId), aws.StringValue(s.AvailabilityZone))
+				return fmt.Sprintf("%s (zone: %s, zone-id: %s)", aws.StringValue(s.SubnetId), aws.StringValue(s.AvailabilityZone), aws.StringValue(s.AvailabilityZoneId))
```

Comment:
I wonder if this should just be a json string to structure it now that we are specifying things like `"Node":{"name":"default"}`. Probably nice to have consistent structure here